From 6b690e70888b69a85c23cdcea58540a1b3d81e48 Mon Sep 17 00:00:00 2001
From: Petr Hodina <phodina@protonmail.com>
Date: Sun, 17 Oct 2021 20:42:28 +0200
Subject: [PATCH] Enable unstable feature.


diff --git a/src/lib.rs b/src/lib.rs
index 9f2e81d..5d42008 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -33,6 +33,8 @@
 //!
 //! println!("{}", "strikethrough".strikethrough());
 //! ```
+#![feature(const_if_match)]
+#![feature(const_generics)]
 #![cfg_attr(not(test), no_std)]
 #![doc(html_logo_url = "https://jam1.re/img/rust_owo.svg")]
 
-- 
2.33.0

