From b642e7969daa3fa7ad29e4d9b10561f2b6aca730 Mon Sep 17 00:00:00 2001
From: Petr Hodina <phodina@protonmail.com>
Date: Sun, 17 Oct 2021 21:56:02 +0200
Subject: [PATCH] Comment out attribute - requires rustc>=1.54.


diff --git a/src/lib.rs b/src/lib.rs
index 62a163b..1ea8bb5 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,4 +1,4 @@
-#![doc = include_str!("../README.md")]
+//#![doc = include_str!("../README.md")]
 
 pub use atty::Stream;
 
-- 
2.33.0

